src/backend/utils/gpmon/gpmon.c

```diff
@@ -246,6 +259,9 @@ void gpmon_qlog_packet_init(gpmon_packet_t *gpmonPacket)
  */
 void gpmon_qlog_query_submit(gpmon_packet_t *gpmonPacket)
 {
+	if (!gpmon_qlog_should_send(gpmonPacket))
+		return;
+
 	struct timeval tv;
 
 	GPMON_QLOG_PACKET_ASSERTS(gpmonPacket);
```

Comment:
how about macros like
```suggestion
#define GPMON_QLOG_PACKET_SHOULD_SEND(gpmonPacket) \
	if (!gpmonPacket->u.qlog.istoplevel & log_min_messages > DEBUG4) \
		return
...
	struct timeval tv;

	GPMON_QLOG_PACKET_ASSERTS(gpmonPacket);
	GPMON_QLOG_PACKET_SHOULD_SEND(gpmonPacket);
```
that placed AFTER GPMON_QLOG_PACKET_ASSERTS macros!

Reply:
Sure, I'll replace that